common/k8stest/util_postgres_app.go

```diff
@@ -0,0 +1,314 @@
+	"github.com/openebs/openebs-e2e/common/e2e_config"
+	"strings"
+	"time"
+
+	batchV1 "k8s.io/api/batch/v1"
+	coreV1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+// PostgresApp represents a PostgreSQL application deployment configuration.
+type PostgresApp struct {
+	Namespace    string     // Kubernetes namespace where the PostgreSQL application is deployed.
+	Pod          coreV1.Pod // Kubernetes Pod running the PostgreSQL application.
+	VolUuid      string     // UUID of the provisioned volume for the PostgreSQL application.
+	ReleaseName  string     // Helm release name for the PostgreSQL application.
+	ReplicaCount int        // Number of replicas for the PostgreSQL StatefulSet.
+	ScName       string     // StorageClass name used for provisioning volumes.
+	Standalone   bool       // Indicates if the PostgreSQL deployment is standalone.
+	PvcName      string     // Name of the PersistentVolumeClaim.
+	PgBench      PgBenchApp // PgBench application configuration for benchmarking.
+}
+
+// PostgresInstallReady checks if the PostgreSQL application is installed and ready.
+func (psql *PostgresApp) PostgresInstallReady() error {
+	logf.Log.Info("checking postgres application to be installed")
+	ready := false
+	counter := 12
+	if psql.Standalone {
+		// List all StatefulSets with label "app.kubernetes.io/name=postgresql" in the namespace.
+		stateful, err := gTestEnv.KubeInt.AppsV1().StatefulSets(psql.Namespace).List(context.TODO(), metav1.ListOptions{LabelSelector: "app.kubernetes.io/name=postgresql"})
+		if err != nil {
+			return err
+		}
+		// Check the status of each StatefulSet to determine if all replicas are ready.
+		for _, ss := range stateful.Items {
+			ready = ss.Status.ReadyReplicas == ss.Status.Replicas &&
+				ss.Status.AvailableReplicas == ss.Status.Replicas
+			logf.Log.Info("StatefulSet",
+				"app", "Postgres",
+				"ready", ready,
+				"name", ss.Name,
+				"availableReplicas", ss.Status.AvailableReplicas,
+				"readyReplicas", ss.Status.ReadyReplicas,
+				"currentReplicas", ss.Status.CurrentReplicas,
+			)
+
+		}
+		for !ready && counter > 0 {
+			pods, err := ListPod(psql.Namespace)
+			if err != nil {
+				return err
+			}
+			for _, pod := range pods.Items {
+				if strings.Contains(pod.Name, psql.ReleaseName) {
+					if pod.Status.Phase == coreV1.PodRunning {
+						ready = true
+						psql.Pod = pod
+						logf.Log.Info("Pod",
+							"app", "Postgres",
+							"ready", ready,
+							"name", pod.Name,
+							"status", pod.Status.Phase,
+						)
+						break
+					}
+					logf.Log.Info("Pod",
+						"app", "Postgres",
+						"ready", ready,
+						"name", pod.Name,
+						"status", pod.Status.Phase,
+					)
+				}
+			}
+			if ready {
+				var pvcName = psql.PvcName
+				logf.Log.Info("Verify volume provision", "pvc name", pvcName, "namespace", psql.Namespace)
+				uuid, err := VerifyVolumeProvision(pvcName, psql.Namespace)
```

Comment:
You need to use trim down version of VerifyVolumeProvision here because VerifyVolumeProvision verify mayastor volume also which will fail if we deploy other openebs engine